### PR TITLE
Updated backup_github a lot

### PR DESCRIPTION
**Pull request by sylvinite**
- Refactored logging to suppress github stdout output
- Backup is now compressed and moved
- Removed implicit ruby dependency
- Improved logging & made exceptions more explicit

### CODE REVIEW

**Comment by ewels** [comment]
Super nice!

File: backup_github.py [hunk 1]
```diff
@@ -34,47 +31,81 @@ def __enter__(self):
...
     """
+    #logfile = open(logfile, "w")
     if not password is None:
         gh = Github(login=user, user=user, password=password)
         repos = gh.repos.list(type='all')
     else:
         gh = Github()
         repos = gh.repos.list(type='all', user=user)
     for repo in repos.all():
+        if not password is None and repo.private is True:
```
Comment (Galithil) [inline backup_github.py]: slightly confusing, does the not applies only to password == None ?

**Comment by Galithil** [comment]
Short of the fact that I'm not sure why we should bother with backuping what's on github, I don't have any complaints with this.

**Comment by ewels** [comment]
Only question I have is about using plaintext passwords on the crontab as discussed at lunch. Maybe better to move into a config file?

**Comment by vezzi** [comment]
looks fine

**Pull request merged**